### DISCUSSION TITLE
Why walletconnect v2 must save history action on datalocal ? and what is purpose of history ?

### DISCUSSION THREAD

**Discussion by GrahamMcBain** [WalletConnect/walletconnect-monorepo#1666] [Developer Support] [answered]
Can you explain clearly what you save on dataLocal Dapp ? I see alot keychain. 

walletconnect library should not Save anything on datalocal. Let developer do that job.

<img width="917" alt="Screen_Shot_2022-11-21_at_17 06 41" src="https://user-images.githubusercontent.com/36456059/203207306-db3b886c-7424-4eb7-ad95-aaa69e863589.png">

**Comment by ganchoradkov** [comment, marked as answer]
Hey @GrahamMcBain, WalletConnect v2 uses the browser's `localStorage` to cache data. This solves a big issue where, if it didn't cache the pairings/sessions, you would need to scan new QR code & pair again every time the page is reloaded.

**Reply by KhangBacoor** [reply]
is there another way to cache the pairings/sessions ? or i wannt create again pairing topic and sesion topic

**Reply by ganchoradkov** [reply]
Hey @KhangBacoor the pairings/sessions are cached by the SDK automatically. 
You can get them by calling
`const pairings = client.core.pairing.getPairings()`
`const sessions = client.session.getAll()`

**Reply by KhangBacoor** [reply]
everything work perfectly! thank you

**Comment by NesredinJemal** [comment]
I need transfer money from my wallet
